### PR TITLE
Pull out basics of the CassandraClientPool blacklist

### PR DESCRIPTION
**Pull request by tboam**
**Goals (and why)**: Ultimate goal is allow us to configure the backoff and retry behaviour in the CassandraClientPool.  First step is refactoring this out to allow things to be customised.  This PR just starts pulling out some blacklist stuff.

**Implementation Description (bullets)**: Hide the blacklist in its own class.

**Concerns (what feedback would you like?)**: I know there's lots more refactorings here, all I'm concerned about is whether I've accidentally changed behaviour.

**Where should we start reviewing?**: CassandraClientPoolImpl

**Priority (whenever / two weeks / yesterday)**: a few days

---
This change is [Reviewable](https://reviewable.io/reviews/palantir/atlasdb/2701)

### CODE REVIEW

File: atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraClientPoolImpl.java [hunk 1]
```diff
@@ -143,8 +143,8 @@ public void shutdown() {
     static final int MAX_TRIES_TOTAL = 6;
     @VisibleForTesting
     volatile RangeMap<LightweightOppToken, List<InetSocketAddress>> tokenMap = ImmutableRangeMap.of();
-    @VisibleForTesting
-    Map<InetSocketAddress, Long> blacklistedHosts = Maps.newConcurrentMap();
+
+    private Blacklist blacklist = new Blacklist();
```
Comment (gsheasby) [inline atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraClientPoolImpl.java]: `final`?

File: atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraClientPoolImpl.java [hunk 2]
```diff
@@ -312,7 +317,7 @@ private void debugLogStateOfPool() {
             StringBuilder currentState = new StringBuilder();
             currentState.append(
                     String.format("POOL STATUS: Current blacklist = %s,%n current hosts in pool = %s%n",
-                    blacklistedHosts.keySet().toString(), currentPools.keySet().toString()));
+                    blacklist.getBlacklistedHosts().keySet().toString(), currentPools.keySet().toString()));
```
Comment (gsheasby) [inline atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraClientPoolImpl.java]: Law of demeter: could potentially create `String getBlacklist()` or have `Blacklist.toString()` return this content.

File: atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraClientPoolImpl.java [hunk 3]
```diff
@@ -229,6 +229,11 @@ public void shutdown() {
         metricsManager.deregisterMetrics();
     }
 
+    @VisibleForTesting
```
Comment (gsheasby) [inline atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraClientPoolImpl.java]: We could mock the blacklist instead.

**Comment by tboam** [comment]
Thanks, will continue with refactor.  I've looked again at the Blacklist and it is really just a list (all the methods are from Collection) with entry time.  I'll probably add a `BlacklistManager` class which might be a good place to test the required behaviour.

**Pull request merged**